### ISSUE TITLE
Image preprocessing (Normalize) cannot be configured — silent metric corruption

### ISSUE THREAD

**Issue by pizzajojo** [CAIIVS/raitap#158]
## Summary

raitap loads image samples as raw `[0.0, 1.0]` tensors (PIL → `ToTensor`) and feeds them straight into the model. There is no way to declare the input preprocessing pipeline in the config schema, and the model loader (`raitap/models/model.py:_build_arch_from_config`) calls the torchvision factory but discards the `Weights` enum's canonical `.transforms()`.

Any model trained with the standard torchvision convention (ImageNet `Normalize(mean, std)`) therefore receives **out-of-distribution inputs** through raitap. Predictions degenerate, and `metrics.json` reports numbers that contradict the model's actual performance — silently, with no warning.

## Affected versions

`raitap==0.5.0` (likely 0.4.x too — not retested).

## Expected vs actual

**Expected**: image classification models trained with `torchvision.transforms.Normalize(imagenet_mean, imagenet_std)` are evaluated with the same preprocessing, either implicitly (using `Weights.transforms()`) or via an explicit config.

**Actual**: raw `[0, 1]` tensors flow into a model that was trained on normalised inputs ≈ `(x - mean) / std`. The model collapses to predicting one class. raitap reports the resulting metrics as authoritative.

One observed case: a binary classifier whose own held-out eval reported accuracy `0.987` was reported by raitap as `{accuracy: 0.625, recall: 1.0, precision: 0.625, f1: 0.769}`. `390 / (390 + 234) = 0.625` matches the class prior — every sample predicted as the positive class.

## Evidence

```
$ grep -rn "Normalize\|IMAGENET\|imagenet_mean" raitap/
(no matches)
```

`raitap/configs/schema.py`, `DataConfig` surface:

```python
@dataclass
class DataConfig:
    name: str = "isic2018"
    description: str | None = None
    source: str | None = None
    forward_batch_size: int | None = None
    input_metadata: dict[str, Any] | None = None    # shape/modality only
    labels: LabelsConfig = field(default_factory=LabelsConfig)
```

`input_metadata` is shape/modality (image vs tabular vs time series), not numeric preprocessing. No field for `Normalize`/`Resize`.

`raitap/models/model.py:_build_arch_from_config`:

```python
weights = "DEFAULT" if pretrained else None
return factory(weights=weights, num_classes=num_classes)
```

The Weights enum is selected only to populate initial parameters — its `.transforms()` is never used.

## Reproduction

```python
# repro.py
import torch, torch.nn as nn
from torchvision.models import resnet18, ResNet18_Weights

model = resnet18(weights=ResNet18_Weights.DEFAULT)
model.fc = nn.Linear(model.fc.in_features, 2)
# (in real use: fine-tune this model with
#   transforms.Normalize(mean=[0.485, 0.456, 0.406],
#                        std =[0.229, 0.224, 0.225]).)
torch.save(model.state_dict(), "resnet18.pt")
```

`repro.yaml`:

```yaml
hardware: "cpu"
model:
  source: "./resnet18.pt"
  arch: "resnet18"
  num_classes: 2
data:
  name: "repro"
  source: "./images/test"
  labels:
    source: "./images/labels.csv"
    id_column: "image"
    column: "label"
    encoding: "index"
metrics:
  _target_: "ClassificationMetrics"
  task: "binary"
  num_classes: 2
  average: "macro"
reporting:
  _target_: "PDFReporter"
  filename: "report.pdf"
  sample_selection: null
```

Provide `images/test/{NORMAL,PNEUMONIA}/*.jpeg` at `(3, 224, 224)` with a matching `labels.csv` (`image,label`).

```
$ raitap --config-dir . --config-name repro
$ cat outputs/<latest>/metrics/metrics.json
```

Metrics do not reflect the model's actual accuracy.

## Suggested fix

torchvision already ships the canonical preprocessing for every supported arch on the `Weights` enum:

```python
from torchvision.models import get_model_weights
weights = get_model_weights(arch)
preprocess = weights.DEFAULT.transforms()
# → Resize + CenterCrop + ToTensor + Normalize(imagenet_mean, imagenet_std)
```

raitap could:

1. Default to `get_model_weights(arch).DEFAULT.transforms()` whenever `model.arch` is a torchvision builder name. Matches the de-facto convention for state-dicts produced by torchvision training scripts.
2. Expose `data.preprocessing` as a manual override for custom pipelines / non-torchvision archs.
3. If the state-dict was fine-tuned against a non-default `Weights` variant (e.g. `ResNet50_Weights.IMAGENET1K_V2`), let the config select it via `model.weights: IMAGENET1K_V2` so the right `.transforms()` is used.

This avoids a new dataclass for the 90% case while keeping escape hatches.

## Severity

Medium-high. Image classification is raitap's primary advertised use case. The mismatch is silent: PDF + metrics file look authoritative but bear no relation to the trained model. Caught only by cross-checking against an independent eval loop.

## Related

`reporting.sample_selection` is required in 0.5.0 but pre-0.5 configs don't have it; raises `ConfigAttributeError` rather than defaulting to `null`. Workaround: `+reporting.sample_selection=null` on the Hydra CLI. May warrant a separate issue if the field should default to null.
